### DISCUSSION TITLE
[eCBIS] Family name - use cases to think about

### DISCUSSION THREAD

**Discussion by HenryRae** [opensrp/fhircore#824] [Ideas]
**Some use cases I think we need to have more thoughts on**
After thinking more about [[eCBIS] Remove the family member #767](https://github.com/opensrp/fhircore/issues/767#issue-1057769582) issue, I came up with some use cases for us to think about and the changes these might happen. If it happens rarely, we can ignore the 1% cases.

What happens to the family details like the family name when:
1. the family head is removed, dies?

- [ ] - when the family head is divorced and moves to create another family, do we create a new family for them with the same family name(remember we using the surname as family name)?
- [ ] - when the family isn't assigned a new family head?

2. another member is assigned the family head role?
3. when the family head dies or is divorced and a new head(maybe a new husband) moves into the family?
4. how would we handle cases when the father/husband has more than one family/compound?

- [ ] - do we register then twice?
- [ ] - do we assign them the family head role in all families he owns?
- [ ] - do we give all the families the same name?

5. how would the CHA differentiate families or family names where the family heads have the same surname?
6. what if a family relocates/moves from one Community to another, and the family head happens to have the same surname as another family

**Comment by f-odhiambo** [comment]
@HenryRae @pld converting this to a Discussion Item for the short term

**Comment by rowo** [comment]
Been thinking about this. 

We were set to [remove the step to create a family from WCARO/VTS](https://smartregister.atlassian.net/browse/LMH1-132) and streamline so the family name would be defined by the head of household (as proposed by Maimoona to simplify that build). However, the above scenarios makes it clear that combining an individual medical record with a social status like family or household name is not a good idea, especially with how things will work in the real world (Henry shared an anecdote where he could see a Ugandan household/family be referred to as a deceased husband's name until a new husband moves in — which wouldn't be possible if we restrict naming to active client records).

For some context, the idea of a "head of household" or "family head" is not something all implementations have had. I believe it was added with WCARO. The point being it's not something familiar to everywhere and so it might not be something we should force everywhere either.

Another thing that came up is I recently changed the label "family" to "household" to match up better with LMH's CHA/CBIS wording. This made it clearer to me that the family/household name is as much an address as it is a reference point for the family group. If a village doesn't have addresses, you can either give a "local" name (which in LMH is a sequential household ID known only to and defined by the CHA in the paper ledgers), or you can give it a name i.e. "Peter's House/Family". So flexibility, I can see something like "Peter's Family" who lives in the "Big Blue Corner House". 

Combined with the scenarios Henry describe (ex: a person who is head of two different households), I think it makes probably decouple these and have the family name be data associated with the structure, similar to GPS,  address, or household ID. That implies we'll need to be able to create family or household relationships without a head being defined (or at least without the head having social meaning to imply it's the head's family and house). "Head of household" or "family head" then is more of a label we apply to members within the family/household... same as the "primary caregiver" designation.

@mberg you should take a look here too in addition to @pld @f-odhiambo

**Reply by mberg** [reply]
I think there are multiple issues here.

First is the idea of a household. I think Roger's idea of linking it to the address (location) can work (more details below).  This is find for grouping prior together and it allows you to name it something descriptive.

In terms of families we will want to be able to associate family members with each other.  Eg children with a care giver. So we need to think if we were using the head of household for that link. If not then we might have to add a step to do make these associations which may be a pain. The important thing would be liking children to their parents.  I think for the rest though it doesn't make a difference from a health standpoint so just lining them to a common location / address makes sense.

So in summary I think this approach can work.

I would love to be able to associate each household to a location.  This would serve as their address.  I'll create a separate issue but the location could have the following properties.

Name - can be household name. This can remain the same I'd head of household dies
#GPS / geosjon - if we want to link to structure like ij reveal 
- address
- plus code (from google)
- description
- photograph

**Reply by rowo** [reply]
Maybe I'm overthinking this also. We can definitely force the head of household so there's that central family figure connecting people, just having it be the name maybe something we don't do.

In the case of an individual being head of household in two different structures, you have the same issue with an individual being a member of two different families. We aren't going to recreate a new client record in both places — we have to either say they can only exist in one primary place or allow individual records to exist in multiple households and make that flexible. Is it kind of like how a child can be both in a school structure, but also their home structure?

Edit: Matt also just brought up maybe the idea of a family isn't needed. Which I kind of agree with. It's almost just a household roster of individuals. The only relationship needed is if it relates to clinical needs — which is only about children's caregivers. Even mother's birth history can just be entered.

**Reply by HenryRae** [reply]
I agree with linking household names to location because back in the African setting I know in my country... we direct people(if in villages) using other families' house e.g take a right from Mr. Jema's house or the late Jema's house or Mrs. Jema's house. 
Giving room for flexibility like allowing users to edit the household names to what the CHA himself understands or could use as reference. Also allowing switching of the head of households only when removing a person, and this can be automated in some cases e.g when the head moves out of the family or dies thus assigning the next person in line(maybe according to age) as the head of household(still allowing CHA to change this).

Maybe this is not so necessary but how do we currently handle Muslim families? Am assuming Liberia's population is mostly Muslims, and Muslims can have more than one wife. this can be in the same or different communities, this could be an edge case not to spend more time on but thought we could share more around it with time.

**Comment by f-odhiambo** [comment]
Some context here on how we landed to the current implementation of Family in FHIR https://github.com/opensrp/fhircore/issues/276

CC @HenryRae @rowo @pld @mberg @dubdabasoduba @maimoonak

**Comment by dubdabasoduba** [comment]
**Representing a Household in OpenSRP FHIRCore using the Location & Patient resource**
----

**Adding a family/household.** 

- The family registration QuestionnaireResponse will be extracted to a **Location resource**. This location resource will have the following attributes
    - An **official** `identifier` *
    - A **secondary** `identifier` 
    - `name` * – This will be the name of the family e.g Matt Berg’s Family
    - `type` * – This will have the code [PTRES](https://www.hl7.org/fhir/v3/RoleCode/cs.html#v3-RoleCode-PTRES) which represents the Patient’s residence
    - `physicalType` * - This identifies the location as a building (we could use room too). The list can be found [here](https://www.hl7.org/fhir/valueset-location-physical-type.html). 
    - `position` – If GPS data is collected. Might necessitate a GPS picker for questionnaires. 
  
**Adding a family member who is the family head**
- The family member registration QuestionnaireResponse will be extracted to a **Patient resource**. This resource will have the following important attributes
     - `tag` * – This will hold a Family Head tag object
     - `Location Extension` * – This extension will hold a reference to the location created above. 

**Adding a family member who is not the family head.**
- The family member registration QuestionnaireResponse will be extracted to a patient resource. This resource will have the following important attributes
    - `link`* – This will hold a reference to the patient resource for the family head created above.
    - `Location Extension` * – This extension will hold a reference to the location created above. 
    - The `link` attribute will be used on instances we want to show all the related person information for a certain patient

**Listing families**
- The OpenSRP FHIRCore app or any other 3rd party application hitting the HAPI FHIR instance will have to query for all the locations with the type [`PTRES`](https://www.hl7.org/fhir/v3/RoleCode/cs.html#v3-RoleCode-PTRES) and `physicalType` **building** or **room** and are `active`.  
- The result of this query is what will be presented and the family register.  

**Listing family members**
**Option 1**
- The OpenSRP FHIRCore app or any other 3rd party application hitting the HAPI FHIR instance will have to query for all the patients with the location Id referenced on the location extension on the patient resources.
- Deceased family members will be listed out but all actions on them will be disabled. 

**Option 2**
- The OpenSRP FHIRCore app or any other 3rd party application hitting the HAPI FHIR instance will have to query for the patient with the tag `Family Head` and location extension referencing the selected family (location Id)
- Then perform another search for the patient that has the family head patient resource referenced on the link attribute.
- Deceased family members will be listed out but all actions on them will be disabled. 

**Removing a family**
- By clicking the remove family button on the OpenSRP FHIRCore family profile menus. A remove family form will be launched.  After filling out the form a QuestionnaireResponse and Encounter will be generated. 
- The location will then be set to inactive.
- The location extension on the patient resources tied to this family (Location Resource) will be removed.  
- We have 2 options on how to handle the patient in family
- **Option 1**
    - Deactivate all the patients in the family. 
    - This means the patients will not appear in any other registers that are health vertical related. E.g if a patient is in the ANC health vertical then they will not appear on the ANC health vertical register mainly because they would be disabled. 
- **Option 2**
    - Leave the patients active
    - This means the family patients can appear in any other health vertical registers and they can still receive care.
Open Question — Do we need to remove the reference to the family head from all the other patient resources? 

**Removing a family head**
- By clicking the remove family button on the OpenSRP FHIRCore family member profile menus. A remove family member form will be launched.  After filling out the form a QuestionnaireResponse and Encounter will be generated.
- A screen to select a new family head will be presented to the users. This list will only show patients above 15yrs.
- A new family head will be selected and the following will happen. 
    - The Family Head tag will be removed from the family member patient resource
    - The patient resource will be set to inactive & deceased (if the reason for removal is death) or left active for the patient to be visible in other health vertical-specific registers. 
    - All the references to the old family head in the other family members will be updated to reference the newly selected family head.
    - The location extension will be removed from the old family head patient resource.
- If no one is old enough to be a family head the following will happen
    - The family head will not be removed or set inactive. We might add another tag to show that they need to be replaced as the family head. We can also use the tag to disable any family related actions on this patient. 
     - When the time comes to select a new family head the change family head function described below will kick in.
 
**Removing family members**
- By clicking the remove family button on the OpenSRP FHIRCore family member profile menus. A remove family member form will be launched.  After filling out the form a QuestionnaireResponse and Encounter will be generated.
- The patient resource will be set to inactive & deceased (if the reason for removal is death) or left active for the patient to be visible in other health vertical-specific registers. 

**Changing family head**
**Changing an inactive family head**
- An inactive family head is a family head who was removed from the family early but could not be completely removed because no one in the family was old enough to assume the family head role. 
- When a family member is finally ready to assume responsibility then the patient resource will be set to inactive & deceased (if the reason for removal is death) or left active for the patient to be visible in other health vertical-specific registers. 
- The Family Head tag will be removed from the family member's patient resource.
- All the references to the old family head in the other family members will be updated to reference the newly selected family head.
- The location extension will be removed from the old family head patient resource.

**Changing an active family head**
- The Family Head tag will be removed from the family member's patient resource.
- All the references to the old family head in the other family members will be updated to reference the newly selected family head.
- A reference link to the new family head will be added to the patient resource for the old family head.

**Comment by dubdabasoduba** [comment]
**Feedback from the CPG Call**
- Instead of using a Tag to represent the Family head we use an extension. 
- We can use the group resource to represent the family which gives us 2 advantages
    -  Querying for members becomes easy
    - Using the `managingEntity` and `RelatedPerson` resources we can identify who the Family head is.

**Reply by joiskash** [reply]
@dubdabasoduba  Thank you so much for this write up.

I was just going through this and was wondering how the RelatedPerson resource will be used. 

Eg: In the context of a Pregnant Woman being the primary Patient in a family of four . The HOH being the eldest in the family (eg: pregnant woman's father), the pregnant woman, her partner and a child.

1. Head of house hold will be represented by a RelatedPerson Resource (Apart from the resource that represents them eg: Person / Patient) i.e. The eldest in the family. I am guessing the identifier will have the logical id of the Person/Patient resource.
2. RelatedPerson.[patient](https://www.hl7.org/fhir/relatedperson-definitions.html#RelatedPerson.patient) will be used to refer to the Patient i.e. the Pregnant woman.
3. Group.managingEntity will refer to the RelatedPerson.
4. Group.members.entity will contain references to all four members of the family.

The question I have is that since RelatedPerson can only be used between two people, how will the child be identified ? Is it through the Group.members resource? Or will there be another RelatedPerson resource connecting the child and mother? or something else?

Similarly how will the partner of the Pregnant woman be connected?
